lib/sqlalchemy/sql/functions.py

```diff
@@ -33,10 +36,14 @@
 from .selectable import FromClause
 from .selectable import Select
 from .visitors import VisitableType
+from .. import exc as sa_exc
 from .. import util
 
 
+_RegistryList = partial(util.defaultdict, list)
+
 _registry = util.defaultdict(dict)
+_case_sensitive_reg = util.defaultdict(_RegistryList)
```

Comment:
**mike bayer** (zzzeek) wrote:

Let's call this "_case_sensitive_registry" for some naming reasons below.

Also with nested defaultdicts I just name them all at once for clarity:

   _case_sensitive_registry = util.defaultdict(
        lambda: util.defaultdict(dict)
    )

it looks like functools.partial is a C implementation now but the Python version illustrates there is a lot of overhead to calling it, especially for what it is.

Reply:
Ok for the name.

I was not aware of this overhead, thanks for the details.